src/main/java/xyz/nucleoid/extras/integrations/relay/ChatRelayIntegration.java

```diff
@@ -123,59 +128,56 @@ private void tick(MinecraftServer server) {
         }
     }
 
-    private ActionResult onSendChatMessage(ServerPlayerEntity player, String content) {
+    private void onSendChatMessage(MessageSender sender, String content) {
         var body = new JsonObject();
 
-        var profile = player.getGameProfile();
-
         var senderRoot = new JsonObject();
-        senderRoot.addProperty("id", profile.getId().toString());
-        senderRoot.addProperty("name", profile.getName());
+        senderRoot.addProperty("id", sender.uuid().toString());
+        senderRoot.addProperty("name", sender.name().getString());
 
         body.add("sender", senderRoot);
         body.addProperty("content", content);
 
         this.chatSender.send(body);
 
-        return ActionResult.PASS;
     }
 
     private void broadcastMessage(MinecraftServer server, ChatMessage message) {
         var playerManager = server.getPlayerManager();
 
         var sender = message.getSenderName();
-        var prefix = new LiteralText("<@").append(sender).append(">").formatted(Formatting.GRAY);
+        var prefix = Text.literal("<@").append(sender).append(">").formatted(Formatting.GRAY);
         var result = new MessageBuilder(prefix);
 
         if (message.replyingTo != null) {
             result.append(this.createReplyText(message.replyingTo));
         }
 
         for (var line : message.lines) {
-            result.append(new LiteralText(line));
+            result.append(Text.literal(line));
         }
 
         if (message.attachments != null) {
             for (var attachment : message.attachments) {
-                result.append(new LiteralText("[Attachment: " + attachment.name + "]").styled(style ->
+                result.append(Text.literal("[Attachment: " + attachment.name + "]").styled(style ->
                     style.withFormatting(Formatting.BLUE, Formatting.UNDERLINE)
                             .withClickEvent(new ClickEvent(ClickEvent.Action.OPEN_URL, attachment.url))
-                            .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new LiteralText("Open attachment")))
+                            .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Text.literal("Open attachment")))
                 ));
             }
         }
 
-        playerManager.broadcast(result.build(), MessageType.CHAT, Util.NIL_UUID);
+        playerManager.broadcast(result.build(), MessageType.CHAT);
```

Comment:
This will get double chat decoration (though missing sender)- would be best to send this as `MessageType.SYSTEM`